sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuite.scala

```diff
@@ -36,21 +38,23 @@ import org.apache.spark.unsafe.types.UTF8String
 /**
  * Test suite for data type casting expression [[Cast]].
  */
-class CastSuite extends SparkFunSuite with ExpressionEvalHelper {
+abstract class CastSuiteBase extends SparkFunSuite with ExpressionEvalHelper {
```

Comment:
Can you talk about the rationale of this test refactor? I would keep the test suite unchanged, and just add a few new test cases for `AnsiCastSuite`.

One day if we have two totally different cast (like `AnsiCast` and `SafeCast`), then I would refactor the test suite annd test these 2 casts separately.

Reply:
See the comment in https://github.com/apache/spark/pull/25997#discussion_r330434072
There are cases that will fail with `AnsiCast`.

Reply:
The tests create `Cast` and check it with ansi mode on and off. How is it related to `AnsiCast`?

Reply:
My point is, I don't think `AnsiCast` worth a separated suite. It's exactly the same with `Cast` and ansi=true, and we have many tests for it already,

Reply:
I think we should fully test it for the future development in the new expression.
For the overflow test cases, we can reduce duplicated code in this way.